specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/edgeMachineJob.tsp

```diff
@@ -0,0 +1,761 @@
+  @visibility(Lifecycle.Read)
+  startTimeUtc?: string;
+
+  @doc("End time in utc of step.")
+  @visibility(Lifecycle.Read)
+  endTimeUtc?: string;
+
+  @doc("Status of step. Allowed values are 'Error', 'Success', 'InProgress'")
+  @visibility(Lifecycle.Read)
+  status?: string;
+
+  @doc("List of nested steps during job execution..")
+  @visibility(Lifecycle.Read)
+  @identifiers(#["name"])
+  steps?: Step[];
+
+  @doc("List of exceptions in job execution.")
+  @visibility(Lifecycle.Read)
+  exception?: string[];
+
+  @doc("Error information for the step")
+  @visibility(Lifecycle.Read)
+  error?: Azure.ResourceManager.Foundations.ErrorDetail;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Operating system profile.")
+model OsProvisionProfile {
+  @doc("Name of the operating system.")
+  osName?: string;
+
+  @doc("Type of the operating system.")
+  osType?: string;
+
+  @doc("Version of the operating system.")
+  osVersion?: string;
+
+  @doc("Location of the operating system image.")
+  osImageLocation?: string;
+
+  @doc("Validated Solution Recipe version to be used for the job")
+  vsrVersion?: string;
+
+  @doc("Hash of the OS package downloaded")
+  imageHash?: string;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/secret-prop" "GPG Public Key is not sensitive information"
+  @doc("GPG Public Key used for package verification")
+  gpgPubKey?: string;
+
+  @doc("Operation sub type of OS Provisioning")
+  operationType?: OSOperationType = OSOperationType.Provision;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("Operating system profile.")
+model DownloadOsProfile {
+  @doc("Name of the operating system.")
+  osName?: string;
+
+  @doc("Type of the operating system.")
+  osType?: string;
+
+  @doc("Version of the operating system.")
+  osVersion?: string;
+
+  @doc("Location of the operating system image.")
+  osImageLocation?: string;
+
+  @doc("Validated Solution Recipe version to be used for the job")
+  vsrVersion?: string;
+
+  @doc("Hash of the OS package downloaded")
+  imageHash?: string;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/secret-prop" "GPG Public Key is not sensitive information"
+  @doc("GPG Public Key used for package verification")
+  gpgPubKey?: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("OS Provision Operation type")
+union OSOperationType {
+  string,
+
+  @doc("OS Provisioning operation")
+  Provision: "Provision",
+
+  @doc("OS Update operation")
+  Update: "Update",
+
+  @doc("OS ReImage operation")
+  ReImage: "ReImage",
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("User configuration.")
+model UserDetails {
+  @doc("Name of the user.")
+  userName: string;
+
+  @doc("Type of the secret used for authentication. Change it to enum in public preview")
+  secretType: string;
+
+  @doc("Location of the secret used for authentication.")
+  secretLocation?: string;
+
+  @doc("Location of the secret used for authentication.")
+  sshPubKey?: string[];
```

Comment:
>sshPubKey [](http://example.com/codeflow?start=2&length=9)

needs to be plural

Reply:
At least fix the description. Apparently it's a collection of keys?